### PR TITLE
added support for multiple sort fields

### PR DESCRIPTION
**Pull request by ferronrsmith**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
   "author": "Daniel Paz-Soldan <daniel.pazsoldan@gmail.com>",
   "contributors": [
     "Nicolás Fantone",
-    "Nauval Atmaja"
+    "Nauval Atmaja",
+    "Ferron Hanse"
```
Comment (danpaz) [inline package.json]: 👍 👍 👍

**Comment by danpaz** [comment]
My preference is to keep the api surface as small as possible, but I hear your need for being able to pass in an array of pre-formed sort clauses. Maybe we can add support for arrays in the `sort` method instead of creating a new `sorts` method. 

I'm going to think on this one for a bit, but also open to discussion if you or anyone else disagrees.

**Comment by danpaz** [comment]
> append if the sort field and direction is passed in

👍 

> overwrite in an array is passed

What about concat if an array is passed? Maybe something like this:

```js
sort(field, direction = 'asc') {
  this._body.sort = this._body.sort || []

  if (_.isArray(field)) {
    this._body.sort.concat(field)
  } else {
    this._body.sort.concat([{[field]: order: direction}])
 }
```

**Comment by ferronrsmith** [comment]
Perfect :+1: :+1: :+1:

**Comment by danpaz** [comment]
Cool! I guess one other thing worth mentioning is, we probably don't need the `sort` property to be an array if there's only one clause. 

If you feel good about modifying the PR to work with this approach, go for it!

**Comment by ferronrsmith** [comment]
it's easier to manage chaining with the proposed implementation

```javascript
builder.sort('version').sort('categories', 'desc');
```

It doesn't really matter to ES (only the order is important.

**Comment by ferronrsmith** [comment]
Should it handle errors or ignore duplicates like ES ?

**Comment by danpaz** [comment]
> Should it handle errors or ignore duplicates like ES ?

That sounds great to me, it's the kind of common sense error handling I would like to see implemented in bodybuilder to guide the user to building well-formed queries. But to keep this PR focused you can probably can defer this for now.

**Comment by danpaz** [comment]
@ferronrsmith Please let me know when this is ready for review, not sure if you had planned to do more on this.

**Comment by eilgin** [comment]
I need to use this feature so it would be great if you could review it 👍

**Comment by ferronrsmith** [comment]
@danpaz review and let me know thanks.

**Comment by danpaz** [comment]
Sorry for the delay! I'll make some comments inline.

File: src/index.js [hunk 2]
```diff
@@ -90,12 +90,40 @@ class BodyBuilder {
    * @returns {BodyBuilder} Builder class.
    */
   sort(field, direction = 'asc') {
-    this._body.sort = {
-      [field]: {
-        order: direction
+    this._body.sort = this._body.sort || [];
+    var self = this;
```
Comment (danpaz) [inline src/index.js]: I think you won't need to stash a reference to this if you use arrow functions on L104 and L105

File: src/index.js [hunk 3]
```diff
@@ -90,12 +90,40 @@ class BodyBuilder {
...
+          }
+      } else {
+          this._body.sort = [this._body.sort];
       }
+
+      sortMerge(this._body.sort, field, direction);
+
     }
-    return this
+    return this;
```
Comment (danpaz) [inline src/index.js]: We've been using no semi colons throughout. Optionally you can add an eslint rule for that, but not required for this PR.

File: src/utils.js [hunk 4]
```diff
@@ -55,3 +55,29 @@ export function boolMerge(newObj, currentObj, boolType = 'and') {
...
+  });
+
+  if(idx == -1) {
+    current.push(payload);
+  } else {
+    _.extend(current[idx], payload);
+  }
+
+  return current;
+}
```
Comment (danpaz) [inline src/utils.js]: Add new line to end of file

File: src/index.js [hunk 5]
```diff
@@ -90,12 +90,40 @@ class BodyBuilder {
...
+    if (_.isArray(field)) {
+
+        if(_.isPlainObject(this._body.sort)) {
+            this._body.sort = [this._body.sort];
+        }
+
+        if(_.isArray(self._body.sort)) {
+            _.each(field, function (sorts) {
+                _.each(sorts, function(value, key) {
+                    sortMerge(self._body.sort, key, value);
```
Comment (danpaz) [inline src/index.js]: Would this still work if I did 
```
.sort([{
  "channel": {
    "order": "desc"
  }
}])
```
We probably need to support this syntax as well.

**Comment by danpaz** [comment]
Looking great! I added some comments, mostly about style and a few edge cases. Could you also update the readme with the new array syntax for `sort`?

**Comment by ferronrsmith** [comment]
awesome great!

**Comment by ferronrsmith** [comment]
updated and squashed

**Comment by danpaz** [comment]
Looks good @ferronrsmith thanks for the great PR!

**Pull request merged**